### PR TITLE
Display localized gun ammo type name

### PR DESCRIPTION
**Pull request by BrettDong**
type

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -986,7 +986,7 @@ std::string item::info( bool showtext, std::vector<iteminfo> &info ) const
                                                   _( ammo_name( mod->ammo_type() ).c_str() ) ), mod->ammo_capacity(), true );
             }
         } else {
-            info.emplace_back( "GUN", _( "Type: " ), ammo_name( mod->ammo_type() ) );
+            info.emplace_back( "GUN", _( "Type: " ), _( ammo_name( mod->ammo_type() ).c_str() ) );
```
Comment (codemime) [inline src/item.cpp]: `ammunition_type::name()` is a far better place to put the localization.

File: src/item.cpp [hunk 2]
```diff
@@ -981,10 +981,10 @@ std::string item::info( bool showtext, std::vector<iteminfo> &info ) const
             if( mod->ammo_capacity() ) {
                 info.emplace_back( "GUN", _( "<bold>Capacity:</bold> " ),
                                    string_format( ngettext( "<num> round of %s", "<num> rounds of %s", mod->ammo_capacity() ),
-                                                  _( ammo_name( mod->ammo_type() ).c_str() ) ), mod->ammo_capacity(), true );
+                                                  ammo_name( mod->ammo_type() ).c_str() ), mod->ammo_capacity(), true );
             }
         } else {
-            info.emplace_back( "GUN", _( "Type: " ), ammo_name( mod->ammo_type() ) );
+            info.emplace_back( "GUN", _( "Type: " ), ammo_name( mod->ammo_type() ).c_str() );
```
Comment (codemime) [inline src/item.cpp]: Minor: there's no need for `.c_str()`, it only adds unnecessary conversions: `std::string` -> c string -> `std::string`.

**Comment by codemime** [comment]
There's another [one missed translation](https://github.com/CleverRaven/Cataclysm-DDA/pull/20001/files/0735abc3fdffcf7c0499548a44b739e76626f888#diff-f7ca61cf749eba039da961e8a7be1630L901).

**Comment by BrettDong** [comment]
> There's another one missed translation.

Which line?

**Comment by codemime** [comment]
901. See the like (though it doesn't work well, a github issue I suppose).

**Comment by BrettDong** [comment]
Yeah, that doesn't work well. The diff directly begins at line 941.

File: src/item.cpp [hunk 3]
```diff
@@ -981,10 +981,10 @@ std::string item::info( bool showtext, std::vector<iteminfo> &info ) const
             if( mod->ammo_capacity() ) {
                 info.emplace_back( "GUN", _( "<bold>Capacity:</bold> " ),
                                    string_format( ngettext( "<num> round of %s", "<num> rounds of %s", mod->ammo_capacity() ),
-                                                  ammo_name( mod->ammo_type() ).c_str() ), mod->ammo_capacity(), true );
```
Comment (codemime) [inline src/item.cpp]: This `c_str()` (unlike the one I've mentioned) is still required for `string_format()`.

File: src/item.cpp [hunk 4]
```diff
@@ -981,7 +981,7 @@ std::string item::info( bool showtext, std::vector<iteminfo> &info ) const
             if( mod->ammo_capacity() ) {
                 info.emplace_back( "GUN", _( "<bold>Capacity:</bold> " ),
                                    string_format( ngettext( "<num> round of %s", "<num> rounds of %s", mod->ammo_capacity() ),
-                                                  _( ammo_name( mod->ammo_type() ).c_str() ) ), mod->ammo_capacity(), true );
```
Comment (codemime) [inline src/item.cpp]: This `c_str()` (unlike the one I've mentioned) is still required for `string_format()`.

**Comment by BrettDong** [comment]
Uhm, to my shame, committed a stupid mistake again.

**Comment by codemime** [comment]
No worries, @BrettDong.

**Pull request merged**